sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -659,27 +672,18 @@ private[cosmos] class CosmosRowConverter(
         }
     }
 
-    private def getFullFidelityMetadata(objectNode: ObjectNode): Option[ObjectNode] = {
-      if (objectNode == null) {
-        None
-      } else {
-        val metadata = objectNode.get(FullFidelityChangeFeedMetadataPropertyName)
-        if (metadata != null && metadata.isObject) {
-          Some(metadata.asInstanceOf[ObjectNode])
-        } else {
-          None
-        }
+    //  Some of these APIs can be clubbed together to avoid code duplication.
+    //  However, I don't think it is worth losing code readability and code simplicity.
+    private def getAttributeNode(objectNode: ObjectNode, attributeName: String): JsonNode = {
+      objectNode.get(attributeName) match {
+        case jsonNode: JsonNode => jsonNode
+        case _ => null
```

Comment:
Don't use null in Scala - it is considered a No-Go (Option[T]) is used instead.

I don't actually think you need this helper method. objectNode.get("name") is all this one is doing (when using Option instead of null)

But objectNode.get just returns a JsonNode - could be anything from Object to Array - that is why before we had separate helper methods also validating the expected return type.

You can switch to moving objectNode.get instead but need to add validation that the returned JsonNode is of the expected type

Reply:
got it, will do.